addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -6618,6 +6618,38 @@ registry.ImageTools = ImageHandlerOption.extend({
     // Options
     //--------------------------------------------------------------------------
 
+    /**
+     * Toggle image with certain shapes into 0/0(original) and 1:1 ratio.
+     *
+     * @see this.selectClass for parameters
+     */
+    async removeStretch(previewMode, widgetValue, params) {
+        this.trigger_up("disable_loading_effect");
+        const restoreCursor = preserveCursor(this.$target[0].ownerDocument);
+        const img = this._getImg();
+        const document = this.el.ownerDocument;
+        const imageCropWrapperElement = document.createElement("div");
+        // During temporary mounting, making sure component isn't visible.
+        imageCropWrapperElement.classList.add("d-none");
+        document.body.append(imageCropWrapperElement);
+        const imageCropWrapper = await attachComponent(this, imageCropWrapperElement, ImageCrop, {
+            activeOnStart: true,
+            media: img,
+            mimetype: this._getImageMimetype(img),
+        });
+        await imageCropWrapper.component.mountedPromise;
+        widgetValue
+            ? await imageCropWrapper.component.reset()
+            : await imageCropWrapper.component.cropSquare(true);
+        if (isGif(this._getImageMimetype(img))) {
+            img.dataset[img.dataset.shape ? "originalMimetype" : "mimetype"] = "image/png";
+        }
+        await this._reapplyCurrentShape();
+        imageCropWrapperElement.remove();
+        imageCropWrapper.destroy();
```

Comment:
Code suggestion (I posted the whole function instead of just the diff, as it easier):
```js
/**
 * Toggles the ratio of an image with specific shapes between 0:0 (no crop)
 * and 1:1, in order to make it appear squared when needed.
 *
 * @see this.selectClass for parameters
 */
async removeStretch(previewMode, widgetValue, params) {
    this.options.wysiwyg.odooEditor.historyPauseSteps();
    this.trigger_up("disable_loading_effect");
    const img = this._getImg();
    const document = this.el.ownerDocument;
    const stretch = !!widgetValue;

    const imageCropWrapperElement = document.createElement("div");
    // During temporary mounting, making sure the component isn't visible.
    imageCropWrapperElement.classList.add("d-none");
    document.body.append(imageCropWrapperElement);
    const imageCropWrapper = await attachComponent(this, imageCropWrapperElement, ImageCrop, {
        activeOnStart: true,
        media: img,
        mimetype: this._getImageMimetype(img),
    });
    await imageCropWrapper.component.mountedPromise;

    // Reset the ratio if the image should be stretched, crop it into a
    // square otherwise.
    if (stretch) {
        await imageCropWrapper.component.reset();
    } else {
        await imageCropWrapper.component.cropSquare(false);
        if (isGif(this._getImageMimetype(img))) {
            img.dataset[img.dataset.shape ? "originalMimetype" : "mimetype"] = "image/png";
        }
    }
    imageCropWrapper.destroy();
    imageCropWrapperElement.remove();
    await this._reapplyCurrentShape();
    this.trigger_up("enable_loading_effect");
    this.options.wysiwyg.odooEditor.historyUnpauseSteps();
},
```